src/Blackboard.cs

```diff
@@ -73,6 +73,21 @@ public void RegisterAndSet<T>(string id, T item)
             Set(id, item);
         }
 
+        public IEnumerable<KeyValuePair<string, object>> GetAll_Debug()
+        {
+            // splice in the types as default objects
+            var dataCopy = new Dictionary<string, object>(data);
+            foreach (var kvp in types)
+            {
+                if (!dataCopy.ContainsKey(kvp.Key))
```

Comment:
I would move that into the foreach definition instead.

Like 

```
foreach (var kvp in types.Where(kvp => !dataCopy.ContainsKey(kvp.Key))) {
                dataCopy[kvp.Key] = kvp.Value.IsValueType ? Activator.CreateInstance(kvp.Value) : null;
            }
```

Reply:
Honestly I don't personally like that as much :V I think my personal expectation here is that the contents of a foreach parameter is mostly non-mutating as the iteration happens; like, you could add a `.ToList()` to the end without breaking anything. But in this case we expect the contents of the things the foreach iterates over to kinda mutate over time based on the body of the loop, and that's *sort of weird*.

This might be a weird constraint that I have, but it feels right.